### ISSUE TITLE
Sftp backup

### ISSUE THREAD

**Issue by FuNFooKaS** [HDInnovations/UNIT3D-Community-Edition#3395]
### UNIT3D Version

7.2.0

### PHP & Platform

8.3.1 ubuntu0.22.04.1

### Laravel verion

10.39.0

### Have you done this?

- [X] I am willing to share my stack trace and logs
- [ ] I can suggest a fix as a Pull Request

### Expectation

php artisan backup:run
to run full backup  and send to  server that is configured in the config file,this has worked for a long time just noticed since i updated to v7 that it is not

### Description

the  error  below

### Stack trace & logs

```shell
php artisan backup:run
Starting backup...

   Error

  Class "League\Flysystem\PhpseclibV3\SftpConnectionProvider" not found

  at vendor/laravel/framework/src/Illuminate/Filesystem/FilesystemManager.php:215
    211▕      * @return \Illuminate\Contracts\Filesystem\Filesystem
    212▕      */
    213▕     public function createSftpDriver(array $config)
    214▕     {
  ➜ 215▕         $provider = SftpConnectionProvider::fromArray($config);
    216▕
    217▕         $root = $config['root'] ?? '/';
    218▕
    219▕         $visibility = PortableVisibilityConverter::fromArray(

      +5 vendor frames

  6   [internal]:0
      Spatie\Backup\BackupDestination\BackupDestinationFactory::Spatie\Backup\BackupDestination\{closure}()
      +18 vendor frames

  25  artisan:35
      Illuminate\Foundation\Console\Kernel::handle()
```

i try and run this  composer require league/flysystem-sftp-v3
./composer.json has been updated
Running composer update league/flysystem-sftp-v3
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - league/flysystem-sftp-v3[2.4.0, ..., 2.5.0] require league/flysystem ^2.0.0 -> found league/flysystem[2.0.0, ..., 2.5.0] but the package is fixed to 3.23.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - league/flysystem-sftp-v3[3.0.0, ..., 3.22.0] require phpseclib/phpseclib ^3.0 -> found phpseclib/phpseclib[3.0.0, ..., 3.0.35] but the package is fixed to 2.0.46 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires league/flysystem-sftp-v3 * -> satisfiable by league/flysystem-sftp-v3[2.4.0, ..., 2.5.0, 3.0.0, ..., 3.22.0].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
You can also try re-running composer require with an explicit version constraint, e.g. "composer require league/flysystem-sftp-v3:*" to figure out if any version is installable, or "composer require league/flysystem-sftp-v3:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
### Are you a UNIT3D sponsor?

No, don't give priority to this

**Comment by HDVinnie** [comment]
Install `composer require league/flysystem-sftp-v3 "^3.0"` if want sftp. https://laravel.com/docs/10.x/filesystem#sftp-driver-configuration

**Comment by FuNFooKaS** [comment]
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - league/flysystem-sftp-v3[3.0.0, ..., 3.22.0] require phpseclib/phpseclib ^3.0 -> found phpseclib/phpseclib[3.0.0, ..., 3.0.35] but the package is fixed to 2.0.46 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires league/flysystem-sftp-v3 ^3.0 -> satisfiable by league/flysystem-sftp-v3[3.0.0, ..., 3.22.0].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

**Comment by FuNFooKaS** [comment]
`{
    "name": "laravel/laravel",
    "description": "The Laravel Framework.",
    "keywords": [
        "framework",
        "laravel"
    ],
    "license": "MIT",
    "type": "project",
    "require": {
        "php": ">=8.2",
        "ext-dom": "*",
        "ext-json": "*",
        "ext-libxml": "*",
        "ext-zend-opcache": "*",
        "ext-zip": "*",
        "assada/laravel-achievements": "^2.6",
        "bjeavons/zxcvbn-php": "^1.3.1",
        "doctrine/dbal": "^3.7.2",
        "gabrielelana/byte-units": "^0.5.0",
        "guzzlehttp/guzzle": "^7.8.1",
        "hdvinnie/laravel-html-purifier": "^2.0.0",
        "hdvinnie/laravel-joypixel-emojis": "^2.0.0",
        "hdvinnie/laravel-security-headers": "^2.0.0",
        "intervention/image": "^2.7.2",
        "joypixels/assets": "^v7.0.1",
        "laravel/fortify": "^1.19.1",
        "laravel/framework": "^10.37.3",
        "laravel/tinker": "^2.8.2",
        "league/flysystem-sftp": "^3.22",
        "livewire/livewire": "^2.12.6",
        "marcreichel/igdb-laravel": "^3.8.1",
        "paragonie/constant_time_encoding": "^2.6.3",
        "spatie/laravel-backup": "^8.4.1",
        "spatie/laravel-cookie-consent": "^3.2.4",
        "spatie/laravel-image-optimizer": "^1.7.1",
        "spatie/ssl-certificate": "^2.6.2",
        "symfony/dom-crawler": "^6.4.0",
        "theodorejb/polycast": "dev-master",
        "voku/anti-xss": "^4.1.42",
        "vstelmakh/url-highlight": "^3.0.3"
    },
    "require-dev": {
        "brianium/paratest": "7.2.2",
        "fakerphp/faker": "^1.23",
        "jasonmccreary/laravel-test-assertions": "^2.3",
        "laravel/pint": "^1.13.7",
        "laravel/sail": "^1.26.3",
        "mockery/mockery": "^1.6.7",
        "nunomaduro/collision": "7.7.0",
        "nunomaduro/larastan": "^2.7.0",
        "pestphp/pest": "^2.9.0",
        "pestphp/pest-plugin-drift": "^2.0",
        "pestphp/pest-plugin-laravel": "^2.1",
        "pestphp/pest-plugin-livewire": "^2.0",
        "phpunit/phpunit": "10.2.3",
        "spatie/laravel-ignition": "^2.3.1"
    },
    "config": {
        "preferred-install": "dist",
        "sort-packages": true,
        "optimize-autoloader": true,
        "platform": {
            "ext-mcrypt": "1.0"
        },
        "allow-plugins": {
            "pestphp/pest-plugin": true
        }
    },
    "extra": {
        "laravel": {
            "dont-discover": []
        }
    },
    "autoload": {
        "psr-4": {
            "App\\": "app/",
            "Database\\Factories\\": "database/factories/",
            "Database\\Seeders\\": "database/seeders/"
        },
        "files": [
            "app/Helpers/Helpers.php"
        ]
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "minimum-stability": "stable",
    "prefer-stable": true,
    "scripts": {
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi",
            "@php artisan vendor:publish --force --tag=livewire:assets --ansi",
            "@php artisan vendor:publish --tag=public --provider=\"hdvinnie\\LaravelJoyPixels\\LaravelJoyPixelsServiceProvider\""
        ],
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate --ansi"
        ],
        "post-update-cmd": [
            "@php artisan vendor:publish --tag=laravel-assets --ansi --force"
        ]
    }
}
`

**Comment by HDVinnie** [comment]
installed fine for me on two instances. do `composer update` and then `composer require league/flysystem-sftp-v3 "^3.0"`

**Comment by FuNFooKaS** [comment]
`composer require league/flysystem-sftp-v3 "^3.0"
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Continue as root/super user [yes]?
./composer.json has been updated
Running composer update league/flysystem-sftp-v3
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - league/flysystem-sftp-v3[3.0.0, ..., 3.22.0] require phpseclib/phpseclib ^3.0 -> found phpseclib/phpseclib[3.0.0, ..., 3.0.35] but the package is fixed to 2.0.46 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires league/flysystem-sftp-v3 ^3.0 -> satisfiable by league/flysystem-sftp-v3[3.0.0, ..., 3.22.0].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
`

**Comment by FuNFooKaS** [comment]
all good now i had to edit  the composer.json  and add `"league/flysystem-sftp-v3": "^3.0",`  the update
